Build failure with GHC-8.4.4

```
src/lib/Data/Variant/Excepts.hs:13:14: error:
    Unsupported extension: BlockArguments
   |
13 | {-# LANGUAGE BlockArguments #-}
   |           
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,7 +17,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        ghc: ['9.10.1', '9.8.2', '9.6.5', '9.4.7', '9.2.8', '9.0.2', '8.10.7']
+        ghc: ['9.10.1', '9.8.2', '9.6.5', '9.4.7', '9.2.8', '9.0.2', '8.10.7', '8.8.4', '8.6.5', '8.4.4']
 
     env:
       CONFIG: "--enable-tests --enable-benchmarks"

--- a/cabal.project
+++ b/cabal.project
@@ -1,0 +1,6 @@
+packages: .
+
+constraints:
+  -- required until a new version of mwc-random is released
+  -- see: https://github.com/haskell/mwc-random/pull/89
+  random < 1.3

--- a/src/lib/Data/Variant/Excepts.hs
+++ b/src/lib/Data/Variant/Excepts.hs
@@ -10,7 +10,6 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE AllowAmbiguousTypes #-}
-{-# LANGUAGE BlockArguments #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
@@ -178,8 +177,8 @@ instance (Monad m) => Monad (Excepts es m) where
 #if MIN_VERSION_base(4,12,0)
 instance (MonadFail m) => MonadFail (Excepts es m) where
 #endif
-   {-# INLINABLE fail #-}
-   fail = Excepts . fail
+    {-# INLINABLE fail #-}
+    fail = Excepts . fail
 
 instance MonadTrans (Excepts e) where
     {-# INLINABLE lift #-}
@@ -401,7 +400,7 @@ runBothE ::
    ( KnownNat (Length (b:e2))
    , Monad m
    ) => (forall x y. m x -> m y -> m (x,y)) -> Excepts e1 m a -> Excepts e2 m b -> Excepts (Tail (Product (a:e1) (b:e2))) m (a,b)
-runBothE exec f g = Excepts do
+runBothE exec f g = Excepts $ do
    (v1,v2) <- exec (runE f) (runE g)
    pure (veitherProduct v1 v2)
 

--- a/src/lib/Data/Variant/Tuple.hs
+++ b/src/lib/Data/Variant/Tuple.hs
@@ -19,6 +19,10 @@
 {-# LANGUAGE StandaloneKindSignatures #-}
 #endif
 
+#if !MIN_VERSION_GLASGOW_HASKELL(8,6,0,0)
+{-# LANGUAGE TypeInType #-}
+#endif
+
 -- | Tuple helpers
 module Data.Variant.Tuple
    ( uncurry3

--- a/variant.cabal
+++ b/variant.cabal
@@ -1,6 +1,6 @@
 cabal-version:       2.4
 name:                variant
-version:             1.0
+version:             1.0.1
 synopsis:            Variant and EADT
 license:             BSD-3-Clause
 license-file:        LICENSE

